simSPI/helpers.py

```diff
@@ -0,0 +1,308 @@
+    """
+    if filename is not None:
+        xyz = get_xyz_from_pdb(filename)
+        distance = pdist(xyz[0, ...])
+        return np.amax(distance)
+    return None
+
+
+def get_xyz_from_pdb(filename=None):
+    """Get particle coordinates from .pdb file.
+
+    Parameters
+    ----------
+    filename : str
+        Relative path to file containing topological information of particle
+    """
+    if filename is not None:
+        traj = md.load(filename)
+        atom_indices = traj.topology.select("name CA or name P")
+        traj_small = traj.atom_slice(atom_indices)
+        return traj_small.xyz
+    return None
+
+
+def write_crd_file(
+    numpart,
+    xrange=np.arange(-100, 110, 10),
+    yrange=np.arange(-100, 110, 10),
+    crd_file="crd.txt",
+):
+    """Write particle data to .crd file.
+
+    Parameters
+    ----------
+    numpart : int
+        Number of particles.
+    xrange : ndarray
+        Valid horizontal range for display.
+    yrange : ndarray
+        Valid vertical range for display.
+    crd_file : str
+        Relative path to output .crd file.
+    """
+    if os.path.exists(crd_file):
+        print(crd_file + " already exists.")
+    else:
+        rotlist = get_rotlist(numpart)
+        with open(crd_file, "w") as crd:
+            crd.write("# File created by TEM-simulator, version 1.3.\n")
+            crd.write("{numpart}  6\n".format(numpart=numpart))
+            crd.write(
+                "#            \
+                x             \
+                y             \
+                z             \
+                phi           \
+                theta         \
+                psi  \n"
+            )
+            i = 0
+            for y in yrange:
+                for x in xrange:
+                    if i == int(numpart):
+                        break
+                    crd_table = {
+                        "x": x,
+                        "y": y,
+                        "z": 0,
+                        "phi": rotlist[i][0],
+                        "theta": rotlist[i][1],
+                        "psi": rotlist[i][2],
+                    }
+                    crd.write(
+                        "{0[x]:14.4f} \
+                        {0[y]:14.4f} \
+                        {0[z]:14.4f} \
+                        {0[phi]:14.4f} \
+                        {0[theta]:14.4f} \
+                        {0[psi]:14.4f}\n".format(
+                            crd_table
+                        )
+                    )
+                    i += 1
+
+
+def get_rotlist(numpart):
+    """Return a rotation list containing Euler angles.
+
+    Parameters
+    ----------
+    numpart : int
+        Number of particles.
+    """
+    rotlist = []
+    for x in range(0, numpart + 1):
+        x = special_ortho_group.rvs(3)
+        y = rotation_matrix_to_euler_angles(x)
+        rotlist.append(y)
+    return rotlist
+
+
+def rotation_matrix_to_euler_angles(R):
```

Comment:
`rotation_matrix_to_euler_angles` is both here and in `crd.py`. Remove all duplicate code. Just import it from the same place instead of writing it again.